test/system/src/main/java/io/pravega/test/system/framework/services/MarathonBasedService.java

```diff
@@ -71,7 +73,9 @@ public boolean isRunning() {
             GetAppResponse app = marathonClient.getApp(this.id);
             log.debug("App Details: {}", app);
             //app is not running until the desired instance count is equal to the number of task/docker containers
-            if (app.getApp().getTasksRunning().intValue() == app.getApp().getInstances().intValue()) {
+            // and the state of application is healthy.
+            if ((app.getApp().getTasksRunning().intValue() == app.getApp().getInstances().intValue())
+                    && app.getApp().getTasksRunning().intValue() == app.getApp().getTasksHealthy().intValue()) {
```

Comment:
Will getTasksHealthy() alone be sufficient ?

Reply:
yes, we are ensuring that the taskrunning.count and task.healthy.count matches the desired instance.count.